vochain/state.go

```diff
@@ -482,24 +610,39 @@ func (v *State) EnvelopeExists(processID, nullifier []byte, isQuery bool) (bool,
 	return e != nil, nil
 }
 
-// iterateProcessID iterates fn over state tree entries with the processID prefix.
+// iterateVotes iterates fn over state tree entries with the processID prefix.
 // if isQuery, the IAVL tree is used, otherwise the AVL tree is used.
-func (v *State) iterateProcessID(processID []byte,
-	fn func(key []byte, value []byte) bool, isQuery bool) bool {
+func (v *State) iterateVotes(processID []byte,
+	fn func(vid []byte, sdbVote *models.StateDBVote) bool, isQuery bool) error {
 	v.RLock()
 	defer v.RUnlock()
-	if isQuery {
-		v.Store.ImmutableTree(VoteTree).Iterate(processID, fn)
-	} else {
-		v.Store.Tree(VoteTree).Iterate(processID, fn)
+	votesTree, err := v.mainTreeViewer(isQuery).DeepSubTree(
+		[]*statedb.TreeConfig{ProcessesCfg, VotesCfg.WithKey(processID)})
+	if err != nil {
+		return err
 	}
-	return true
+	var callbackErr error
```

Comment:
+100 to giving this variable a name different than `err`, because it helps clarify its purpose and avoid typos